control-plane-operator/controllers/awsprivatelink/awsprivatelink_controller.go

```diff
@@ -299,51 +325,102 @@ func reconcileAWSEndpointService(ctx context.Context, awsEndpointService *hyperv
 			return fmt.Errorf("endpoint %s not found, resetting status", serviceName)
 		}
 		log.Info("endpoint exists", "endpointID", endpointID)
-		return nil
+		endpointDNSEntries = output.VpcEndpoints[0].DnsEntries
+	} else {
+		if !hasAWSConfig(&hcp.Spec.Platform) {
+			return fmt.Errorf("AWS platform information not provided in HostedControlPlane")
+		}
+		input := &ec2.CreateVpcEndpointInput{
+			ServiceName:     aws.String(serviceName),
+			VpcId:           aws.String(hcp.Spec.Platform.AWS.CloudProviderConfig.VPC),
+			VpcEndpointType: aws.String(ec2.VpcEndpointTypeInterface),
+			SubnetIds:       []*string{hcp.Spec.Platform.AWS.CloudProviderConfig.Subnet.ID},
+		}
+		output, err := ec2Client.CreateVpcEndpointWithContext(ctx, input)
+		if err != nil {
+			return err
```

Comment:
Worth wrapping this error with some kind of context about what we were doing at the time and the input values?

Reply:
The error from AWS is already pretty verbose.  I think it would capture the cause of the failure.